smtp/message.js

```diff
@@ -309,34 +441,38 @@ class MessageStream extends Stream {
 			fs.open(attachment.path, 'r', opened);
 		};
 
+		/**
+		 * @param {MessageAttachment} attachment the metadata to use as headers
+		 * @param {function(): void} callback the function to call after output is finished
+		 * @returns {void}
+		 */
 		const output_stream = (attachment, callback) => {
 			if (attachment.stream.readable) {
-				let previous = null;
+				let previous = Buffer.alloc(0);
```

Comment:
don't know the overhead of opting for `previous` to always be a `Buffer`; opted for it to monomorphize the type & allow for a simple check against `byteLength` later on.

Reply:
+1. impact is likely trivial for the use case.